source/transactions/transactions.rst

```diff
@@ -744,24 +753,71 @@ format:
 
     { ok : 1, writeConcernError: {code: <Number>, errmsg : "..."} }
 
-Mongos Pinning for Sharded Transactions
----------------------------------------
+Sharded Transactions
+--------------------
+
+MongoDB 4.2 (maxWireVersion 8) introduces support for sharded transactions.
+Sharded transactions support all of the same features as single replica set
+transaction but introduce two new driver concepts: mongos pinning and the
+``recoveryToken`` field.
 
-The server supports sharded transactions starting in MongoDB 4.2 (
-maxWireVersion 8). The server requires that drivers MUST send all commands for
-a single transaction to the same mongos.
+Mongos Pinning
+~~~~~~~~~~~~~~
+
+Drivers MUST send all commands for a single transaction to the same mongos
+(excluding retries of commitTransaction and abortTransaction).
 
 After the driver selects a mongos for the first command within a transaction,
 the driver MUST pin the ClientSession to the selected mongos. Drivers MUST
-send all commands that are part of the same transaction, including
-commitTransaction and abortTransaction and any retries thereof, to the
-same mongos.
+send all subsequent commands that are part of the same transaction (excluding
+retries of commitTransaction and abortTransaction) to the same mongos.
+
+When to unpin
+^^^^^^^^^^^^^
+
+Drivers MUST unpin a ClientSession when a command within a transaction
+fails with a TransientTransactionError (including a server selection error).
```

Comment:
Isn't "(including a server selection error)" redundant in light of the description of a [TransientTransactionError](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#transienttransactionerror)? This includes network errors, server selection failures, and any other errors the server chooses to label.

Separate question: do we avoid discussing UnknownTransactionCommitResult here because the subsequent paragraph talks about unpinning as soon as a commit/abort is issued (i.e. before such a label would even be assigned)?

Reply:
Fixed the "server selection error" redundancy. 

Do we need to unpin after an UnknownTransactionCommitResult error?  I wrote this thinking that the unpinning rule would also apply to the commitTransaction commands run as part of the commit helper.

Reply:
> apply to the commitTransaction commands run as part of the commit helper.

Are you referring to `ClientSession.commitTransaction()` or the `withTransaction()` helper? I'm not sure it really matters, as the paragraph below this says:

> Additionally, Drivers MUST unpin a ClientSession after running any commitTransaction or abortTransaction attempt regardless if that attempt succeeded or failed.

That surely covers both cases.

Reply:
In response to [this conversation with Esha](https://github.com/mongodb/specifications/pull/471#discussion_r258627327), I've updated this section so that drivers remain pinned after successful commits.